pkg/queue/concurrency_state.go

```diff
@@ -85,3 +88,55 @@ func ConcurrencyStateHandler(logger *zap.SugaredLogger, h http.Handler, pause, r
 		<-done
 	}
 }
+
+// Pause sends to an endpoint when request counts drop to zero
+func Pause(endpoint string) error {
```

Comment:
The layering with passing `endpoint` in here seems a little odd. It feels like it should be captured in the functions passed to `ConcurrencyStateHandler` so it itself wouldn't care what to pass to these functions.